scylla-proxy/src/proxy.rs

```diff
@@ -1174,6 +1175,17 @@ impl ProxyWorker {
     }
 }
 
+// Returns next free last IP address octet for another proxy instance.
+// Useful for concurrent testing.
+#[doc(hidden)]
+pub fn get_exclusive_address_last_octet() -> u8 {
+    static ADDRESS_LAST_OCTET: AtomicU16 = AtomicU16::new(42);
+    ADDRESS_LAST_OCTET
+        .fetch_add(1, Ordering::Relaxed)
+        .try_into()
+        .expect("Loopback address pool for tests depleted")
+}
+
```

Comment:
I guess you are using a 16-bit atomic here to detect overflow?

I think this function unnecessarily restricts itself to just one byte. AFAIK the loopback subnet is quite large - the first octet has to be 127 and the rest is fairly arbitrary, so we have 2^24 - 2 addresses to choose from. While it is realistic to deplete a pool of `256 - 42` addresses in tests, with U32 we won't have to worry about overflow.

Reply:
I also think this function should just return a whole address to make life simpler to the callers.